Yosemite/YosemiteTests/Stores/ProductStoreTests.swift

```diff
@@ -604,7 +604,7 @@ final class ProductStoreTests: XCTestCase {
         let storageProduct1 = viewStorage.loadProduct(siteID: sampleSiteID, productID: sampleProductID)
         XCTAssertEqual(storageProduct1?.toReadOnly(), sampleProductMutated())
         XCTAssertEqual(viewStorage.countObjects(ofType: Storage.Product.self), 2)
-        XCTAssertEqual(viewStorage.countObjects(ofType: Storage.ProductTag.self, matching: NSPredicate(format: "siteID == %lld", sampleSiteID)), 5)
+        XCTAssertEqual(viewStorage.countObjects(ofType: Storage.ProductTag.self, matching: NSPredicate(format: "siteID == %lld", sampleSiteID)), 10)
```

Comment:
The reason is that initially the tags are stored in the wrong way, in fact, I did some manual tests, and I noticed that for some reason there where some duplicated tags (every parameter was duplicated), but the tag was associated with a different product. Basically, the updates, inserts, or prunes of the provided StorageProduct's tags using the provided read-only Product's tags weren't working correctly, and sometimes was possible to see a tag multiple times in the database if it was connected to multiple Products. Quite strange things happened.

I solved it this PR, in the `ProductStore` updating the `handleProductTags:` method, and now the counts are correct, because the `sampleProduct()` method generates 9 tags:
```
let tag1 = ProductTag(siteID: siteID, tagID: 37, name: "room", slug: "room")
        let tag2 = ProductTag(siteID: siteID, tagID: 38, name: "party room", slug: "party-room")
        let tag3 = ProductTag(siteID: siteID, tagID: 39, name: "30", slug: "30")
        let tag4 = ProductTag(siteID: siteID, tagID: 40, name: "20+", slug: "20")
        let tag5 = ProductTag(siteID: siteID, tagID: 41, name: "meeting room", slug: "meeting-room")
        let tag6 = ProductTag(siteID: siteID, tagID: 42, name: "meetings", slug: "meetings")
        let tag7 = ProductTag(siteID: siteID, tagID: 43, name: "parties", slug: "parties")
        let tag8 = ProductTag(siteID: siteID, tagID: 44, name: "graduation", slug: "graduation")
        let tag9 = ProductTag(siteID: siteID, tagID: 45, name: "birthday party", slug: "birthday-party")
```
and the `sampleProductMutated()` generates 5 tags, but some of these with the same site ID and tag id.

```
        let tag1 = ProductTag(siteID: siteID, tagID: 37, name: "something", slug: "something")
        let tag2 = ProductTag(siteID: siteID, tagID: 38, name: "party room", slug: "party-room")
        let tag3 = ProductTag(siteID: siteID, tagID: 39, name: "3000", slug: "3000")
        let tag4 = ProductTag(siteID: siteID, tagID: 45, name: "birthday party", slug: "birthday-party")
        let tag5 = ProductTag(siteID: siteID, tagID: 95, name: "yep", slug: "yep")
```

After that the `handleProductTags:` is called the second time (when a new `sampleProductMutated()` is added), the final count is 10 (4 tags updated, and one added).